core/src/components/radio/radio.md.scss

```diff
@@ -70,12 +70,19 @@
 
 // Material Design Radio: Disabled
 // -----------------------------------------
+// The radio itself should use the disabled
+// opacity set by its spec, while the label
+// should match the other form controls
 
-// .item-md.item-radio-disabled ion-label {
-:host(.radio-disabled) {
+:host(.legacy-radio.radio-disabled),
+:host(.radio-disabled) .label-text-wrapper {
   opacity: $radio-md-disabled-opacity;
 }
 
+:host(.radio-disabled) .native-wrapper {
+  opacity: $radio-md-icon-disabled-opacity;
+}
```

Comment:
I might be losing it because I just took another look at the Material Design 2 & 3 specs for radio and both use 38% opacity. I have no idea where I saw it was supposed to be `0.63`. I'll update this and checkbox to use the same as label and then update the global variable from `0.4` to `0.38`.

Reply:
Okay I just realized why I did that. Setting the opacity to `0.63` gets it be the proper color because it's *on top of* the already opaque background color (radio uses `rgb(0, 0, 0, 0.60)` as its color). We would need to change the color back to 100% `#000` in order for the `0.38` to be correct. So `0.63` is the number I got when I modified it to match the spec.

We could change it to something like this:

```css
:host(.radio-disabled) {
  --color: rgb($text-color-rgb, 1.0);

  opacity: $radio-md-disabled-opacity;
}
```

or we could only change the color and not set the opacity for the icon (this seems to be how MD does it):

```css
:host(.radio-disabled) .native-wrapper {
  --color: rgb($text-color-rgb, $radio-md-disabled-opacity);
}
```

But using `0.38` on the entire radio (label + icon) results in the colors being different from each other (which is wrong):

![image](https://github.com/ionic-team/ionic-framework/assets/6577830/9e1f6cd2-6f3f-4c3d-8c46-e0c87976d884)

If you hover over them with digital color meter you'll see the label is `#9F9F9F` and the icon is `#C6C6C6`. These should both be the same `#9F9F9F`.

Reply:
Alternatively, we can leave this as is and create a separate issue to fix the checkbox & radio disabled colors, plus investigate if range or toggle share the same issue. This might be outside of the scope of making the labels match.

Reply:
Brandy and I discussed this offline. We are going to move forward with the `0.63` value and add comments as to how that was chosen. I think us using opacities with background colors is something we should reconsider in the future, as it seems to just cause issues for us. I'd like to do design work in the future and get some consensus.

Reply:
Discussed with Liam offline, we agreed to just add the `0.63` opacity back in with comments as to why and in the future we can re-evaluate if we want to change the color of the checkbox/radio. Updated in this commit https://github.com/ionic-team/ionic-framework/pull/27396/commits/82d3e883965958ea2dcdd04f8961bc6ec75df519 and comments added here https://github.com/ionic-team/ionic-framework/pull/27396/commits/07a66cd94e104f6986620fd469deb70405f64b02